Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -50,6 +50,10 @@ extension SQLiteHistory: HistoryRecommendations {
             "       ORDER BY visitDate DESC" +
             "   )" +
             "   LEFT JOIN \(TableHistory) ON \(TableHistory).id = s" +
+            "   INNER JOIN (" +
+            "       SELECT \(TableHistory).domain_id AS domain_id, MAX(\(TableHistory).id) AS history_id FROM \(TableHistory)" +
+            "       GROUP BY \(TableHistory).domain_id" +
+            "   ) AS domains ON domains.domain_id = \(TableHistory).domain_id AND \(TableHistory).id = history_id" +
```

Comment:
If you use the date, you'd need to join again to get the contributing grouped value, or sub select with an order and limit.